File: java/MaxTreeDepth.java
class Solution {

    /**
     * Function that returns the maximum depth of a binary tree.
     * @param root the starting node.
     * @return the maximum depth of the tree.
     */
    public int maxDepth(TreeNode root) {

        //If the root of the tree is null just return 0, because the max depth is 0.
        if(root == null){
            return 0;
        }

        //Create leftDepth and rightDepth variables that call maxDepth recursively
        int leftDepth = maxDepth(root.left);
        int rightDepth = maxDepth(root.right);

        //Use the max function to determine which tree is deeper, +1 to include root.
        return Math.max(leftDepth, rightDepth) + 1;
    }
}

File: python/MaxTreeDepth.py
from typing import Optional

from python.TreeNode import TreeNode

class Solution:

    # Function that returns the maximum depth of a binary tree.
    def maxDepth(self, root: Optional[TreeNode]) -> int:

        # Base case null tree.
        if root is None:
            return 0
        
        # Since depth = height, need to analyze the left and right subtrees. 
        left_depth = self.maxDepth(root.left)
        right_depth = self.maxDepth(root.right)

        # Return biggest height, +1 to account for root node. 
        return max(left_depth, right_depth) + 1